### PR TITLE
issue-7: modify all tests to be in pure YAML rather than raw ruby

### PR DESCRIPTION
**Pull request by ppeble**
Fixes https://github.com/holidays/definitions/issues/7

Title says it all. What do we think? I would appreciate a spot check that I didn't leave any obvious mistakes. I automated the generation of the new YAML based off of the old ruby but...there was still a lot of manual intervention. I feel that the chances for errors are high.

I'm not going to merge this for a bit. Before I do that I need to get a branch up on the Ruby side to generate our tests from this YAML rather than just shoving the raw ruby that exists today. I don't know how long that will take (with my life circumstances, baby and all that).

I'm happy with how this turned out! 😄

### CODE REVIEW

**Comment by ppeble** [comment]
Note for myself: I need to update the `SYNTAX` doc to reflect the new 'tests' rules.

**Comment by hrach** [comment]
Hi, I've written https://github.com/sygic-travel/holidays-calculator-php (currently in early stages).
The language agnostic tests would be awsome.

Thought, I'm not sure if the proposed format is good one: the test are pretty hard to read and validate, what values they are expecting. Repeating region and formal polute it. Maybe putting the test into region subarray would help. And the formal/informal may have default.

**Comment by ppeble** [comment]
@ttwo32 I would like to merge this in (once I fix the conflicts) so that it becomes our defacto way of doing tests going forward. Fixing conflicts has been very difficult for me and I would prefer that we just merge it, which will force me to finish the changes on the other repo.

The downside to this approach is that we would not be able to do another release until I finish the test generation on the holidays repo but I think I'm basically done. Just fixing minor issues. And I think this is the pressure I need. What do you think?

**Comment by ttwo32** [comment]
I think this is necessary to use these definitions in other languages.
And we need to update the SYNTAX doc to reflect the new 'tests' rules.
But... don't worry!! I will help you to .:)

**Comment by ppeble** [comment]
Great! Thank you @ttwo32. I would ask then that we do _not_ merge any new definitions changes until this one is done. I'm almost done with the final bit of US conflicts and after that I'll make the necessary changes to the SYNTAX (thank you for reminding me!). 

Hopefully I'll have that handled today or tonight, at which point I'll put up the 'final' version of this for you to look over and I'll switch my attention to a release for the latest defs.

**Comment by ppeble** [comment]
Had to do a few merges because the changes were so big. I know this isn't as elegant as cleaner commits but...I'm just so tired of keeping these up to date. 😄 

I'm going to get the release out now and then do the SYNTAX bit.

**Comment by ppeble** [comment]
@ttwo32 Done with SYNTAX update. 👍

**Comment by ppeble** [comment]
@ttwo32 What do you think? Look good in the SYNTAX file?

File: SYNTAX.md [hunk 1]
```diff
@@ -328,26 +328,53 @@ If a user does not specify `observed` when calling the gem then 1/1 will be the
...
+  * regions (required) - an array of strings (NOT symbols). Multiple regions can be passed. Even a single region must be in an array.
+  * options (optional) - an array of options to use for this test. Can be either 'informal' or 'observed'. Must be an array of strings, e.g. `['informal', 'observed']`
+* expect (required)
+  * name (optional) - the name of the holiday you are expecting. Must be a string.
+  * holiday (optional) - a boolean indicating whether the given values result in a holiday. Defaults to 'true' if not present. Must be true or false.
+
+One or the other of the `expect` keys is required. If you do not specify a `name` then you should set `holiday: false`.
+
+Here are some more examples. First example shows multiple dates, multiple regions, additional options, and an expectation that the result will be the named holiday.
+
```
Comment (ttwo32) [inline SYNTAX.md]: These  examples are good.:)

File: SYNTAX.md [hunk 2]
```diff
@@ -328,26 +328,53 @@ If a user does not specify `observed` when calling the gem then 1/1 will be the
 
 ## Tests
 
-All definition files should have tests included.  In the YAML file, tests are just a block of Ruby code:
+All definition files should have tests included. At this time we do not enforce any rules on coverage or numbers of tests.
+However, in general, PRs will not be accepted if they are devoid of tests that cover the changes in question.
 
+The format is a straightforward 'given then expect'. Here is a simple example:
```
Comment (ttwo32) [inline SYNTAX.md]: Good.It,s easy to write test by 3 sections.

**Comment by ttwo32** [comment]
@ppeble Looks good to me!! I really appreciate your hard work.:)

**Comment by ttwo32** [comment]
Woo, Why ruby 2.1.0 test failed?

**Comment by ppeble** [comment]
@ttwo32 I would like your permission to merge this to master, even though the ruby 2.1.0 tests are failing. We know it is already fixed in master and I don't want to have to either rebase or merge a new commit in just to make the PR fully pass.

Other than that...this is good! Basically done and ready to go. Here is the PR to use the new format in the ruby side: https://github.com/holidays/holidays/pull/282

If you are okay with this merge, I will go to master and make sure it builds. Then I will change the ruby PR to point to master. Then we can have final signoff on the ruby side. Thanks!

**Pull request merged**